osde2e/splunk_forwarder_operator_tests.go

```diff
@@ -0,0 +1,230 @@
+		err = resrcs.List(ctx, &clusterRoles)
+		Expect(err).Should(BeNil(), "failed to list clusterroles")
+		found = false
+		for _, clusterRole := range clusterRoles.Items {
+			if strings.HasPrefix(clusterRole.Name, rolePrefix) {
+				found = true
+			}
+		}
+		Expect(found).To(BeTrue(), "unable to find cluster role with prefix %s", rolePrefix)
+
+		ginkgo.By("checking the services exist")
+		for _, serviceName := range serviceNames {
+			err = resrcs.Get(ctx, serviceName, operatorNamespace, &corev1.Service{})
+			Expect(err).Should(BeNil(), "service %s/%s not found", operatorNamespace, serviceName)
+		}
+
+		ginkgo.By("checking the deployment exists and is available")
+		deployment := &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: deploymentName, Namespace: operatorNamespace}}
+		err = wait.For(conditions.New(resrcs).DeploymentConditionMatch(deployment, appsv1.DeploymentAvailable, corev1.ConditionTrue))
+		Expect(err).Should(BeNil(), "deployment %s not available", deploymentName)
+
+		//TODO:  post osde2e-common library add configmap check
+		//checkConfigMapLockfile(h, operatorNamespace, operatorLockFile)
+
+		//TODO:  post osde2e-common library add csv check
+		//checkClusterServiceVersion(h, operatorNamespace, operatorName)
+
+		// TODO: post osde2e-common library add upgrade check
+		//checkUpgrade(helper.New(), "openshift-splunk-forwarder-operator",
+		//	"openshift-splunk-forwarder-operator", "splunk-forwarder-operator",
+		//	"splunk-forwarder-operator-catalog")
+	})
+
+	ginkgo.It("admin should be able to manage SplunkForwarders CR", func(ctx context.Context) {
+		sf := makeMinimalSplunkforwarder("SplunkForwarder", "splunkforwarder.managed.openshift.io/v1alpha1", testsplunkforwarder)
+		err := addSplunkforwarder(ctx, sf, "openshift-splunk-forwarder-operator")
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	ginkgo.It("test SplunkForwarders CR, if exists, should be deleted successfully", func(ctx context.Context) {
+		err := deleteSplunkforwarder(ctx, testsplunkforwarder, operatorNamespace)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	ginkgo.It("dedicated admin should not be able to manage SplunkForwarders CR", func(ctx context.Context) {
+
+		sf := makeMinimalSplunkforwarder("SplunkForwarder", "splunkforwarder.managed.openshift.io/v1alpha1", dedicatedadminsplunkforwarder)
+		err := dedicatedAaddSplunkforwarder(ctx, sf, "openshift-splunk-forwarder-operator")
+		Expect(apierrors.IsForbidden(err)).To(BeTrue())
+		if err == nil {
+			err = deleteSplunkforwarder(ctx, dedicatedadminsplunkforwarder, operatorNamespace)
+			if err != nil {
+				log.Printf("Failed cleaning up %s CR in %s namespace", dedicatedadminsplunkforwarder, operatorNamespace)
+			}
+		}
+	})
+
+})
+
+// Create test splunkforwarder CR definition
+func makeMinimalSplunkforwarder(kind string, apiversion string, name string) sfv1alpha1.SplunkForwarder {
+	sf := sfv1alpha1.SplunkForwarder{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       kind,
+			APIVersion: apiversion,
+		},
```

Comment:
this can be dropped and the params removed

Reply:
Sure! As far as I understand, the TypeMeta only needs to be set if we want to serialize this and persist it to disk (writing it out to yaml for example, allow us to de-serialize it). It will all work for the stuff we are doing just by working with the type.